src/persistent-connections/content.md

```diff
@@ -0,0 +1,465 @@
+
+```slideshow
+{
+  "description": "ClusterIP Service",
+  "slides": [
+    {
+      "image": "assets/iptables-1.svg",
+      "description": "Consider a cluster with three Nodes. Each Node has a Pod deployed."
+    },
+    {
+      "image": "assets/iptables-2.svg",
+      "description": "The light brown Pods are part of a Service. Services don't exist, so the diagram has the component grayed out."
+    },
+    {
+      "image": "assets/iptables-3.svg",
+      "description": "The red Pod wants to issue a request to the Service and eventually reach one of the light brown Pods."
+    },
+    {
+      "image": "assets/iptables-4.svg",
+      "description": "But Service don't exist. There's no process listening on the Service IP address. _How does it work?_"
+    },
+    {
+      "image": "assets/iptables-5.svg",
+      "description": "Before the request is dispatched from the Node, it is intercepted by iptables rules."
+    },
+    {
+      "image": "assets/iptables-6.svg",
+      "description": "The iptables rules know that the Service doesn't exist and proceed to replace the IP address of the Service with one of the IP addresses of the Pods beloning to that Service."
+    },
+    {
+      "image": "assets/iptables-7.svg",
+      "description": "The request has a real IP address as the destination and it can proceed normally."
+    },
+    {
+      "image": "assets/iptables-8.svg",
+      "description": "Depening on your particular network implementation, the request finally reaches the Pod."
+    }
+  ]
+}
+```
+
+_Does iptables use round-robin?_
+
+No, iptables is primarily used for firewalls, and it is not designed to do load balancing.
+
+However, you could [craft a smart set of rules that could make iptables behave like a load balancer](https://scalingo.com/blog/iptables#load-balancing).
+
+And this is precisely what happens in Kubernetes.
+
+If you have three Pods, iptables writes these rules:
+
+1. select Pod 1 as the destination with a probability of 0.33
+1. select Pod 2 as the destination with a probability of 0.33
+1. select Pod 3 as the destination with a probability of 0.33
+
+Since this is a probability, there's no guarantee that Pod 2 is selected after Pod 1 as the destination.
+
+> Iptables use the [statistic module](http://ipset.netfilter.org/iptables-extensions.man.html#lbCD) with `random` mode. So the load balancing algorithm is random.
+
+Now that you're familiar with how Services work let's have a look at more exciting scenarios.
+
+## Keeping it alive
+
+With every request started from the front-end to the backend, a new HTTP connection is opened and closed.
+
+If you front-end makes 100 requests per second to the backend, 100 different HTTP connections are opened and closed in that second.
+
+You can improve the latency and save resources if you open a connection and reuse it for any subsequent requests.
```

Comment:
You can improve the latency and save resources if you open a **TCP** connection and reuse it for any subsequent **HTTP** requests.